src/js/modules/accounts/actions.js

```diff
@@ -158,16 +159,21 @@ export const subscribe = values => {
       type: t.CREATE_SUBSCRIBER_PENDING,
       payload: values,
     });
-    dispatch({ type: t.CLOSE_SUBSCRIPTION_MODAL });
     axios
       .post(`${STUY_SPEC_API_URL}/subscribers`, values, STUY_SPEC_API_HEADERS)
       .then(response => {
         dispatch({
           type: t.CREATE_SUBSCRIBER_FULFILLED,
           payload: response,
         });
-        // Destroys the inputs in the form Subscription
-        dispatch(reset("Subscription"));
+
+        // Lets the user glance at the confirmation message
+        setTimeout(() => {
```

Comment:
Consider using a promise. I like creating a util function that's called delay:

`export const delay = duration =>
  new Promise(
    resolve => setTimeout(resolve, duration)
  );`

Reply:
@jkao1

Reply:
I don't think this is necessary yet. As you often say, don't optimize if it's "for the future". `SubscriptionModal` is our only modal with feedback messages, making it the only use case for such a delay.

Reply:
I'd like us to keep async functionality to the same format though. Mixing promises and callbacks is not advisable. Let's keep it to one style